subcommands/targets/offline-update.go

```diff
@@ -163,6 +187,64 @@ func getTargetInfo(factory string, targetName string) (*ouTargetInfo, error) {
 	return &info, nil
 }
 
+func getWaveTargetMeta(factory string, targetName string, wave string) (*tuf.FileMeta, error) {
+	waveTargets, err := api.WaveTargetsList(factory, true, wave)
+	if err != nil {
+		if herr := client.AsHttpError(err); herr != nil && herr.Response.StatusCode == 404 {
+			return nil, fmt.Errorf("no active Wave with the specified name was found; wave: %s", wave)
+		}
+		return nil, fmt.Errorf("failed to get Wave Target metadata: %s", err.Error())
+	}
+	targets, ok := waveTargets[wave]
+	if !ok {
+		// it looks rather like panic since the call at #281 is supposed to return only the given wave targets
+		subcommands.DieNotNil(fmt.Errorf("no Targets are found for the given wave %s", wave))
+	}
+	if foundTargetMeta, ok := targets.Signed.Targets[targetName]; ok {
+		return &foundTargetMeta, nil
+	} else {
+		return nil, fmt.Errorf("the specified Target is not found among wave targets;"+
+			" target: %s, wave: %s", targetName, wave)
+	}
+}
+
+func getProdTargetMeta(factory string, targetName string, tag string) (*tuf.FileMeta, error) {
+	targets, err := api.ProdTargetsGet(factory, tag, true)
+	if err != nil {
+		if herr := client.AsHttpError(err); herr != nil && herr.Response.StatusCode == 404 {
+			return nil, fmt.Errorf("the specified production Target has not been found; target: %s, tag: %s", targetName, tag)
+		}
+		return nil, fmt.Errorf("failed to get production Target metadata: %s", err.Error())
+	}
+	if foundTargetMeta, ok := targets.Signed.Targets[targetName]; ok {
+		return &foundTargetMeta, nil
+	} else {
+		return nil, fmt.Errorf("no production target with the given tag is found;"+
+			" target: %s, tag: %s", targetName, tag)
+	}
+}
+
+func getCiTargetMeta(factory string, targetName string, tag string) (*tuf.FileMeta, error) {
+	data, err := api.TufMetadataGet(factory, "targets.json", tag, false)
+	if err != nil {
+		if herr := client.AsHttpError(err); herr != nil && herr.Response.StatusCode == 404 {
+			return nil, fmt.Errorf("the specified CI Target has not been found; target: %s, tag: %s", targetName, tag)
```

Comment:
This is actually: no CI targets were found for tag %s.

Reply:
63bac04